eval/src/vespa/eval/eval/aggr.h

```diff
@@ -76,56 +93,69 @@ template <typename T> class Avg {
         _cnt += rhs._cnt;
     };
     constexpr T result() const { return (_sum / _cnt); }
+    static constexpr Aggr enum_type() { return Aggr::AVG; }
 };
 
 template <typename T> class Count {
 private:
     size_t _cnt;
 public:
+    using value_type = T;
     constexpr Count() : _cnt{0} {}
     constexpr Count(T) : _cnt{1} {}
     constexpr void sample(T) { ++_cnt; }
     constexpr void merge(const Count &rhs) { _cnt += rhs._cnt; }
     constexpr T result() const { return _cnt; }
+    static constexpr Aggr enum_type() { return Aggr::COUNT; }
 };
 
 template <typename T> class Prod {
 private:
     T _prod;
 public:
+    using value_type = T;
     constexpr Prod() : _prod{1} {}
     constexpr Prod(T value) : _prod{value} {}
     constexpr void sample(T value) { _prod *= value; }
     constexpr void merge(const Prod &rhs) { _prod *= rhs._prod; }
     constexpr T result() const { return _prod; }
+    static constexpr Aggr enum_type() { return Aggr::PROD; }
+    static T combine(T a, T b) { return (a * b); }
 };
 
 template <typename T> class Sum {
 private:
     T _sum;
 public:
+    using value_type = T;
     constexpr Sum() : _sum{0} {}
     constexpr Sum(T value) : _sum{value} {}
     constexpr void sample(T value) { _sum += value; }
     constexpr void merge(const Sum &rhs) { _sum += rhs._sum; }
     constexpr T result() const { return _sum; }
+    static constexpr Aggr enum_type() { return Aggr::SUM; }
+    static T combine(T a, T b) { return (a += b); }
```

Comment:
is it useful to have `a += b` here?  As it's used from
`*dst = combine(*dst, *src++);`
it gives me some doubt.   I guess you also considered formulation with
`static void combine(T& a, T b) { a += b; }`
and so on.

Reply:
yes, was a relic from the 'past'